emoji-substitution/substitute.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is responsible for performing the logic of replacing
+ * all occurances of each mapped word with its emoji counterpart.
+ */
+
+// emojiMap.js defines the 'dictionary' variable.
+// Referenced here to reduce confusion.
+const emojiMap = dictionary;
+
+/*
+ * For efficiency, create a word --> search RegEx Map too.
+ */
+let regexs = new Map();
+for (let word of emojiMap.keys()) {
+  // We want a global, case-insensitive replacement.
+  // @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+  regexs.set(word, new RegExp(word, 'gi'));
+}
+
+/**
+ * Substitutes emojis into text nodes.
+ * If the node contains more than just text (ex: it has child nodes),
+ * call replaceText() on each of its children.
+ *
+ * @param  {Node} element - The target DOM element.
+ * @return {void}         - Note: the emoji substitution is done inline.
+ */
+function replaceText (element) {
+  // Setting textContent on a node removes all of its children and replaces
+  // them with a single text node. Since we don't want to alter the DOM aside
+  // from substituting text, we only substitute on single text nodes.
+  // @see https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent
+  if (element.nodeType === Node.TEXT_NODE) {
```

Comment:
I was able to reproduce this with `textarea` elements in WordPress editing tools. If I open a published post that has an Except, the text in the Excerpt field has its words replaced. So if I save it for whatever reason, the emojis would end up being published. So, we need to at least discard that type of element.